### PR TITLE
Import third-party JS from commercial-core

### PR DESCRIPTION
**Pull request by chrislomaxjones**
## What does this change?

Remove vendor third party scripts from the commercial code, preferring to use the ones defined in [commercial-core #698](https://github.com/guardian/commercial-core/pull/698).

This PR bumps commercial-core to the latest version that includes these scripts, and two of these vendor scripts are trivial migrations:
- **Pubmatic**
- **Ipsos Mori**

There are some other significant changes for the other two scripts:
- **A9 Apstag**
- **Launchpad** (for Redplanet)

Rather than splitting these scripts out into separate chunks and deferring their execution via a dynamic import, we use versions that are wrapped in functions of the form `() => { ... }`. This enables us to be able to control when we execute the code without having to use a dynamic import.

Note that these chunks weren't obviously split out for performance reasons. Including them will increase the commercial bundle size, but only by a small amount (each of the scripts is ~ 500 bytes parsed size). We could make separate gains by code splitting in different areas of the code if we desired.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?

We shouldn't convert these files to TypeScript. By moving them to a single directory in commercial-core, we should be better able to manage the small amount of JavaScript we will continue to maintain.  

### Tested

- [X] Locally
- [ ] On CODE (optional)

### CODE REVIEW

**Comment by Jakeii** [comment]
if they're moving, could they go straight to `commercial-core`?

**Review by Jakeii — DISMISSED** [review]

**Review by arelra — DISMISSED** [review]
There's also `imr-worldwide` which we've converted but should it also live with other third-party code?

**Comment by chrislomaxjones** [comment]
> if they're moving, could they go straight to commercial-core?

Good shout, assuming we don't mind exporting JS from commercial-core (I can't see a reason not to?)

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-a9.ts [hunk 1]
```diff
@@ -19,24 +20,20 @@ const setupA9 = (): Promise<void | boolean> => {
...
 			commercialFeatures.dfpAdvertising &&
 			!commercialFeatures.adFree &&
 			!window.guardian.config.page.hasPageSkin)
 	) {
-		moduleLoadResult = import(
-			/* webpackChunkName: "a9" */ '../../../../lib/a9-apstag.js'
-		).then(() => {
-			a9.initialise();
+		// Load a9 third party stub
+		a9Apstag();
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-a9.ts:31]: Is the chunking strategy affected? Are we sure this code is not eagerly loaded?

Comment (chrislomaxjones) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-a9.ts:31]: Oops I should've re-marked this as draft. I'll update the description when this is ready, but the intention is that we're changing chunking so these scripts are included in the main bundle (they're only a couple of hundred bytes parsed size each). To defer the scripts execution until the point at which they're dynamically imported (so they don't have side effects until we obtain consent etc), we've wrapped them in `() => { ... }` functions we can call.

**Comment by chrislomaxjones** [comment]
> There's also imr-worldwide which we've converted but should it also live with other third-party code?

I think this is a third party tag rather than vendor JavaScript, but we could consider moving it to commercial-core, we have got a directory for those.

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-a9.ts (hunk 1 above)
Comment (mxdvl) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-a9.ts:31]: Yes, I had a peek at `commercial-core` and realised they were quite small.

I’m still confused why wrapping in an IIFE adds anything over a function, but I’ll let you be the judge of that.

In my understanding, the following three are identical:

```ts
function doThing1() {
  console.log("thing");
}

const doThing2 = () => {
  console.log("thing");
}

const doThing3 = () => {
  (function() { console.log("thing") })();
}
```

Comment (chrislomaxjones) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-a9.ts:31]: So this was the original setup:
```typescript
// stub.js
cosole.log('Do some work');
```

```typescript
// prepare-thing.ts
if (conditionsAreMet) {
  // Top-level console.log won't occur until conditions are met
  await import('./stub.js');
}
```

Which I'm proposing to replace with:
```typescript
// stub.js
const stub = () => {
  console.log('Do some work');
};
```

```typescript
// prepare-thing.ts
import { stub } from './stub.js';

if (conditionsAreMet) {
  // Top-level console.log won't occur until conditions are met
  stub();
}
```

I think the outcome is the same, just with a different chunking strategy.

**Review by Jakeii — APPROVED** [review]

**Pull request merged**